### PR TITLE
Fixed #27602 -- Added Oracle support for BoundingCircle GIS function.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/27602

### CODE REVIEW

**Comment by sir-sigurd** [comment]
buildbot, test on oracle.

File: django/contrib/gis/db/models/functions.py [hunk 1]
```diff
@@ -193,9 +193,11 @@ def __init__(self, expression, relative=False, precision=8, **extra):
         super(AsSVG, self).__init__(*expressions, **extra)
 
 
-class BoundingCircle(GeoFunc):
-    def __init__(self, expression, num_seg=48, **extra):
-        super(BoundingCircle, self).__init__(*[expression, num_seg], **extra)
+class BoundingCircle(OracleToleranceMixin, GeoFunc):
+    def as_postgresql(self, compiler, connection):
+        clone = self.copy()
+        clone.set_source_expressions(self.get_source_expressions() + [Value(self.extra.get('num_set', 48))])
```
Comment (timgraham) [inline django/contrib/gis/db/models/functions.py]: `num_set`looks like a typo for `num_seg`? Is testing of that parameter inadequate to detect the regression? If so, would be nice to add a test for it in a separate PR.

Comment (sir-sigurd) [inline django/contrib/gis/db/models/functions.py]: Yes, you're right. I will create PR to improve `num_seg` testing.

File: docs/ref/contrib/gis/db-api.txt [hunk 2]
```diff
@@ -374,38 +374,38 @@ Database functions
...
-:class:`AsSVG`                        X                             X
-:class:`BoundingCircle`               X
-:class:`Centroid`                     X        X       X            X
-:class:`Difference`                   X        X       X (≥ 5.6.1)  X
-:class:`Distance`                     X        X       X (≥ 5.6.1)  X
-:class:`Envelope`                     X                X            X
+====================================  =======  ==============  ===========  ==========
+Function                              PostGIS  Oracle          MySQL        SpatiaLite
+====================================  =======  ==============  ===========  ==========
+:class:`Area`                         X                        X            X
```
Comment (timgraham) [inline docs/ref/contrib/gis/db-api.txt]: Inadvertently removed Area X for Oracle?

File: docs/ref/contrib/gis/db-api.txt [hunk 3]
```diff
@@ -374,38 +374,38 @@ Database functions
...
+:class:`AsGeoJSON`                    X                                     X
+:class:`AsGML`                        X                                     X
```
Comment (timgraham) [inline docs/ref/contrib/gis/db-api.txt]: I see this change is correct but I wonder if we should make it in a separate commit to ease correcting the docs on 1.10.

File: docs/ref/contrib/gis/db-api.txt [hunk 4]
```diff
@@ -374,38 +374,38 @@ Database functions
...
+:class:`AsKML`                        X                                     X
+:class:`AsSVG`                        X                                     X
+:class:`BoundingCircle`               X        X (≥ 12.1.0.2)
```
Comment (timgraham) [inline docs/ref/contrib/gis/db-api.txt]: Just a though, I wonder if including the version numbers in this chart is a good long term practice considering adjusting the column lengths over time is non-trivial. @claudep, @sir-sigurd opinions?

File: docs/ref/contrib/gis/db-api.txt (hunk 3 above)
Comment (sir-sigurd) [inline docs/ref/contrib/gis/db-api.txt]: I did it by accident too :-).
Fixed it along with the previous issue.

Comment (timgraham) [inline docs/ref/contrib/gis/db-api.txt]: Here: https://github.com/django/django/pull/7697 (for master and 1.10). I'll "revert" it when editing and merging https://github.com/django/django/pull/7695.

File: docs/ref/contrib/gis/db-api.txt (hunk 4 above)
Comment (sir-sigurd) [inline docs/ref/contrib/gis/db-api.txt]: Probably we could make columns have equal width. I think `len('X (≥ 12.1.0.2)'` width ought to be enough for any column.

File: docs/ref/contrib/gis/functions.txt [hunk 5]
```diff
@@ -163,11 +163,20 @@ __ http://www.w3.org/Graphics/SVG/
 
 .. class:: BoundingCircle(expression, num_seg=48, **extra)
 
-*Availability*: `PostGIS <http://postgis.net/docs/ST_MinimumBoundingCircle.html>`__
+*Availability*: `PostGIS <http://postgis.net/docs/ST_MinimumBoundingCircle.html>`__,
+`Oracle (≥ 12.1.0.2) <https://docs.oracle.com/database/121/SPATL/GUID-82A61626-BB64-4793-B53D-A0DBEC91831A.htm#SPATL1554>`_
```
Comment (sir-sigurd) [inline docs/ref/contrib/gis/functions.txt]: I added the link similarly as it done for PostGIS, but not sure if it's good because that link contains Oracle version and probably should be fixed at some point.

Comment (timgraham) [inline docs/ref/contrib/gis/functions.txt]: It's okay for now. If Oracle provides links to the current stable version at some point, it would be great to use them.
